### PR TITLE
PODAUTO-74: Updates for 4.15

### PR DESCRIPTION
**Pull request by jkyros**
Updates for 4.15
-  Update deps
    -   edit (update to 4.15/0.28.3) and run hack/update-vendor.sh
- Update code and build to match updated deps 
- Already at go 1.20 (go.mod, Dockerfile, images/ci/Dockerfile), left it alone
- Rev version from 4.14 -> 4.15:
  ```sed -i 's/4.14/4.15/g' $(git grep -l 4.14 manifests/) images/ci/bundle.Dockerfile hack/manifest-diff-upstream.sh hack/e2e.sh Makefile```
- Verify that Dockerfile.rhel7 & .ci-operator.yaml are up to date (they are. [Thanks, ART team](https://github.com/openshift/vertical-pod-autoscaler-operator/pull/147)!)
- Updated manifests to match upstream
  - also removed workaround for upstream typo from `hack/manifest-diff-upstream.sh`

Note: the controller-runtime bump removed the Namespace field from manager.Options and stuffed that functionality into some cache fields, so I had to adjust that

### CODE REVIEW

File: hack/manifest-diff-upstream.sh [hunk 1]
```diff
@@ -70,7 +70,7 @@ if [ "$NO_DOCKER" = "1" -o -n "$IS_CONTAINER" ]; then
 
   sed -f hack/yamls2list.sed < "$upstream_file" | bin/yaml2json | jq '.items[] | select(.kind=="CustomResourceDefinition" and .metadata.name=="verticalpodautoscalers.autoscaling.k8s.io")' | bin/json2yaml > "$out1"
   # re-add fixed typo (trailing double quote char) to make it match. The jq command can be removed as soon the upstream typo is fixed
```
Comment (joelsmith) [inline hack/manifest-diff-upstream.sh:72]: Looks like this comment can be removed now.

**Comment by jkyros** [comment]
So I tried bumping to controller-runtime v0.16.x and updating our operator's cache configuration to use the new manager cache options but that totally breaks everything, and does not seem to be 1:1 with pre-0.16.z behavior:  

```
2023-11-01T18:36:48-05:00	W1101 23:36:46.434476       1 status.go:226] Operator status degraded: error checking VPA controllers status: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:36:48-05:00	E1101 23:36:47.411344       1 verticalpodautoscaler_controller.go:217] Error reading VerticalPodAutoscalerController: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:36:55-05:00	{"metadata":{"name":"vertical-pod-autoscaler-operator-6b6db69b99.1793a3fff17aff53","namespace":"openshift-vertical-pod-autoscaler","uid":"cf401f29-7255-43cc-8849-54881502286d","resourceVersion":"31290","creationTimestamp":"2023-11-01T23:36:44Z"},"reason":"SuccessfulCreate","message":"Created pod: vertical-pod-autoscaler-operator-6b6db69b99-gbkdq","source":{"component":"replicaset-controller"},"firstTimestamp":"2023-11-01T23:36:44Z","lastTimestamp":"2023-11-01T23:36:44Z","count":1,"type":"Normal","eventTime":null,"reportingComponent":"replicaset-controller","reportingInstance":"","involvedObject":{"kind":"ReplicaSet","namespace":"openshift-vertical-pod-autoscaler","name":"vertical-pod-autoscaler-operator-6b6db69b99","uid":"01ace66a-d814-46d3-abe9-051ee9837c7f","apiVersion":"apps/v1","resourceVersion":"31276","labels":{"k8s-app":"vertical-pod-autoscaler-operator","pod-template-hash":"6b6db69b99"}}}
2023-11-01T18:37:01-05:00	E1101 23:37:01.436224       1 status.go:316] Error getting VerticalPodAutoscalerController: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:01-05:00	W1101 23:37:01.436242       1 status.go:226] Operator status degraded: error checking VPA controllers status: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:16-05:00	E1101 23:37:16.436029       1 status.go:316] Error getting VerticalPodAutoscalerController: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:16-05:00	W1101 23:37:16.436051       1 status.go:226] Operator status degraded: error checking VPA controllers status: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:31-05:00	E1101 23:37:31.436243       1 status.go:316] Error getting VerticalPodAutoscalerController: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:31-05:00	W1101 23:37:31.436278       1 status.go:226] Operator status degraded: error checking VPA controllers status: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:46-05:00	E1101 23:37:46.436412       1 status.go:316] Error getting VerticalPodAutoscalerController: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:37:46-05:00	W1101 23:37:46.436433       1 status.go:226] Operator status degraded: error checking VPA controllers status: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:38:01-05:00	E1101 23:38:01.436331       1 status.go:316] Error getting VerticalPodAutoscalerController: unable to get: /default because of unknown namespace for the cache
2023-11-01T18:38:01-05:00	W1101 23:38:01.436403       1 status.go:226] Operator status degraded: error checking VPA controllers status: unable to get: /default because of unknown namespace for the cache
```
I've moved this down to controller-runtime v0.15.2 until we figure it out, `govulncheck` seems to think that's okay. 

I'll do at least more pass to make sure I didn't miss anything and clean up these commits after I see how the tests go.  This does appear to work locally.  

/hold

File: images/operator-registry/Dockerfile.registry.ci [hunk 2]
```diff
@@ -1,19 +1,19 @@
-FROM quay.io/operator-framework/upstream-registry-builder:v1.13.9 as registry-builder
+FROM quay.io/operator-framework/upstream-registry-builder:v1.30.1 as registry-builder
```
Comment (jkyros) [inline images/operator-registry/Dockerfile.registry.ci:1]: This was like 3 years old, so I bumped it, but if you think I should leave it on `v1.13.9` I will :smile:

**Comment by jkyros** [comment]
Alright, this is using controller-runtime v0.15.2 for now, but it looks like it checks out. 
/hold cancel

**Comment by joelsmith** [comment]
Thanks, sorry for all the cruft. Some tech debt that we should deal with at some point to get us on a better trajectory.
/lgtm

**Pull request merged**